src/Input/Input.js

```diff
@@ -0,0 +1,272 @@
+// @flow weak
+
+import React, { Component, PropTypes } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+import { easing } from '../styles/transitions';
+
+function isDirty(obj) {
+  return obj && obj.value && obj.value.length > 0;
+}
+
+export const styleSheet = createStyleSheet('Input', (theme) => {
+  return {
+    wrapper: {
+      display: 'flex',
+      position: 'relative',
+    },
+    formControl: {
+      marginTop: 10,
+      marginBottom: 10,
+    },
+    inkbar: {
+      '&:after': {
+        backgroundColor: theme.palette.accent.A200,
+        left: 0,
+        bottom: -1,
+        // Doing the other way around crash on IE11 "''"" https://github.com/cssinjs/jss/issues/242
+        content: '""',
+        height: 2,
+        position: 'absolute',
+        width: '100%',
+        transform: 'scaleX(0)',
+        transition: theme.transitions.create(
+          'transform',
+          '200ms',
+          null,
+          easing.easeOut,
+        ),
+      },
+      '&$focused:after': {
+        transform: 'scaleX(1)',
+      },
+    },
+    focused: {},
```

Comment:
I need to use it as a chained selector just above. JSS doesn't automatically create selectors for nested rules like that so you have to declare it at the root.